### ISSUE TITLE
Add outputDir config option to emotion/babel

### ISSUE THREAD

**Issue by ChristopherBiscardi** [emotion-js/emotion#180]
As per #167, I'm using babel-cli to compile packages down into targeted browser ranges and es* versions. Since we can not access the --out-dir config option from babel-cli in babel plugins, the next best option would be to allow an option that could be used as such:

```js
{
  "env": {
    "es6": {
      "plugins": [
        ["emotion/babel", {outputDir: "es6"}]
      ]
    }
  }
}
```

We could then prepend this directory to the output filename when the [file is written](https://github.com/tkh44/emotion/blob/02ab1b140542092d8bfd5b42c8745f1c59563d50/src/babel.js#L347-L366) and npm scripts could use it as such, with all files being put in the right place:

```js
  "scripts": {
    "build:es6": "BABEL_ENV=es6 babel src --out-dir es6"
  },
```

**Comment by vladnicula** [comment]
Any status on this? 

I stumbled upon emotion when researching css in js frameworks and emotion looks the most promising (in terms of package size, dependencies and features). 

This static extraction is crucial to my production build process.

**Comment by tkh44** [comment]
@vladnicula are you doing server side rendering or client only rendering?

**Comment by vladnicula** [comment]
@tkh44 I'm looking to do both. The current project I'm working on is only using client side rendering.

Would be nice to have support for both in the end. 

Btw, opened this https://github.com/emotion-js/emotion/pull/396, it works for my client side project and for my server side bundling in combination with weback extract text plugin. Could not write a proper test because the fs tests are fishy.

**Comment by tkh44** [comment]
I left some comments that may help you out.

**Comment by mitchellhamilton** [comment]
Closed via #583
